### ISSUE TITLE
Error message while using Sardinian (sc) as a browser display language

### ISSUE THREAD

**Issue by asereze** [internetarchive/openlibrary#7105]
### Evidence / Screenshot (if possible)
![Schermata da 2022-10-26 11-38-06](https://user-images.githubusercontent.com/11271324/197993129-0950a452-21bc-400d-a281-a838a9029238.png)

### Relevant url?
https://openlibrary.org/authors/OL1410246A/Raimondo_Carta_Raspi (or any other one)

### Steps to Reproduce

1. Go to any Open Library Page while having Sardinian as a language for displaying pages
2. Do nothing

* Actual: I got an error message and I can't see the page unless I click on the language menu and choose a language from there
* Expected: To be able to see the page without any error message, like it happens with other languages that do not have an Open Library translation (for example Italian, if I use that the page just opens up in English without any issue)

### Details

- **Logged in (Y/N)?** No
- **Browser type/version?** Firefox 105.0
- **Operating system?** Ubuntu 18.04
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints
Add support for Sardinian (I don't know how, I guess it just needs to be added to the list of known locales) to avoid getting that error. If you need informations about the language for that I can provide them.

**Comment by tfmorris** [comment]
I'd suggest a more general solution so that the system falls back gracefully if it gets an unknown language code or one for which a translation doesn't exist.

According to [IANA](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry), there's not only the Sardinian macro language, but an additional four more Sardinian languages (e.g. `sdc` for Sassarese Sardinian)

**Comment by tfmorris** [comment]
That's interesting, but not anything I have control over.

**Comment by asereze** [comment]
Hi everyone, it looks like the issue has been fixed for a while, right now if I use my browser with Sardinian as a language for displaying pages to open any Open Library page it just appears to me in English, with the exception of some elements, like the names of the languages and the date and time, that are translated into Sardinian. The is no error message anymore. Thank you.

Also, related to this, I've created a pull request for Adding a Sardinian translation: #8856
